### PR TITLE
PE-5356: feat(logging)

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
This PR introduces Sentry only for `staging` and `production` environments. Currently, we are just using the `captureException` method, on the `e` method of our `Logger` class.

Other changes:
- remove crashlytics
- refactor of the main file

Added also the step for uploading the source map for Sentry on our workflow.

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/1v38d950rkooo

### CODE REVIEW

**Comment by thiagocarvalhodev** [comment]
TODO:
- Enable Sentry only for production builds
- Code clean up

**Comment by thiagocarvalhodev** [comment]
Refactored the main file and enabled Sentry only on `production` or `staging` builds.

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -12,13 +12,17 @@ env:
...
+  SENTRY_AUTH_TOKEN: ${{secrets.SENTRY_AUTH_TOKEN}}
+  SENTRY_PROJECT: ${{secrets.SENTRY_PROJECT}}
+  
 jobs:
   pre-build:
     uses: ./.github/workflows/test.yml
 
   build-web:
-    needs: pre-build
+    # needs: pre-build
```
[the remark this answers is not in the document]
Comment (karlprieb) [reply .github/workflows/pr.yaml:25]: PR deploys doesn't need sentry configurations as it will not load Sentry

**Review by karlprieb — DISMISSED** [review]

File: lib/main.dart [hunk 2]
```diff
@@ -26,11 +26,10 @@ import 'package:ardrive/utils/pre_cache_assets.dart';
 import 'package:ardrive/utils/secure_key_value_store.dart';
 import 'package:ardrive_http/ardrive_http.dart';
 import 'package:ardrive_io/ardrive_io.dart';
+import 'package:ardrive_logger/ardrive_logger.dart';
```
Comment (karlprieb) [inline lib/main.dart:29]:
```suggestion
```

**Review by karlprieb — APPROVED** [review]

**Review by karlprieb — APPROVED** [review]

**Pull request merged**